### DISCUSSION TITLE
NAudio -- How to stream NAudio to multiple listeners

### DISCUSSION THREAD

**Discussion by wizrd66** [naudio/NAudio#1042] [Q&A]
I am mostly familiar with C# and have done some C and CPP a long time ago.   I was wondering if NAudio would be a gout way to open a music file wav or MP3 most likely and stream it over sockets.   I have sent thing via sockets before, but not music.

the listeners will be PC's or possibly Raspberry PI or similar.    

I am not sure, how to read in the socket data and then re-hydrate it into something that the listener will then feed that to a player like NAudio to play it   It would be nice to be avvle to buffer it and not have to send it in a stream and not all at once.... 

So my solution I create needs to work in a cross platform type situation or have a different player that the Listener  can work with.

I hope that this makes some sensee

**Comment by markheath** [comment]
It is achievable but quite complex to implement well. There is a very basic network streaming demo in the NAudio demo app, showing how you could send compressed audio over the network, decompress it on the receiving end and play it. But for a production ready experience you'd want to use more efficient codecs that NAudio may not support so well, as well as trying to manage buffering correctly to avoid dropouts.

**Comment by wizrd66** [comment]
Yes i predicted needing to do some buffering.  Wasn’t exactly how that
would work . Adding to the end of the stream while playing it .

I was also going to include a position marker so everyone is playing at
the same time.

But you nailed my issue what trying to figure out the buffering how to open
a stream to a port and the receiver would have to handle buffering it etc

Thank you for the reply
